meadow-client/src/Semantics.purs

```diff
@@ -161,61 +131,10 @@ derive instance eqObservation :: Eq Observation
-                                                        , ")"
-                                                        ]
-  show (OrObs observation1 observation2) = joinWith "" [ "OrObs ("
-                                                       , show observation1
-                                                       , ") ("
-                                                       , show observation2
-                                                       , ")"
-                                                       ]
-  show (NotObs observation) = joinWith "" ["NotObs (", show observation, ")"]
-  show (ChoseThis idChoice choice) = joinWith " " [ "ChoseThis"
-                                                  , showIdChoice idChoice
-                                                  , toString choice
-                                                  ]
-  show (ChoseSomething idChoice) = joinWith " " [ "ChoseSomething"
-                                                , showIdChoice idChoice
-                                                ]
-  show (ValueGE value1 value2) = joinWith "" [ "ValueGE ("
-                                             , show value1
-                                             , ") ("
-                                             , show value2
-                                             , ")"
-                                             ]
-  show (ValueGT value1 value2) = joinWith "" [ "ValueGT ("
-                                             , show value1
-                                             , ") ("
-                                             , show value2
-                                             , ")"
-                                             ]
-  show (ValueLT value1 value2) = joinWith "" [ "ValueLT ("
-                                             , show value1
-                                             , ") ("
-                                             , show value2
-                                             , ")"
-                                             ]
-  show (ValueLE value1 value2) = joinWith "" [ "ValueLE ("
-                                             , show value1
-                                             , ") ("
-                                             , show value2
-                                             , ")"
-                                             ]
-  show (ValueEQ value1 value2) = joinWith "" [ "ValueEQ ("
-                                             , show value1
-                                             , ") ("
-                                             , show value2
-                                             , ")"
-                                             ]
-  show TrueObs = "TrueObs"
-  show FalseObs = "FalseObs"
+derive instance genericObservation :: Generic Observation _
 
+instance showObservation :: Show Observation where
+  show o = genericShow o
```

Comment:
Eta reduce - we have done everywhere else for `Show` instances.

Reply:
unfortunately PS fails with a cycle error if you do, can't work out why